tst/EnergyPlus/unit/SurfaceGeometry.unit.cc

```diff
@@ -3274,6 +3274,86 @@ TEST_F(EnergyPlusFixture, SurfaceGeometry_CheckConvexityTest_ASHRAE901_Hospital_
     EXPECT_FALSE(surface.IsConvex);
 }
 
+TEST_F(EnergyPlusFixture, SurfaceGeometry_CheckConvexity_ColinearStability)
+{
+    // Test for #10103 - Regardless of the order of the vertices, we should drop colinear vertices consistently to avoid a fatal error due to vertex
+    // side mismatch
+    //
+    //  y
+    //    ▲          7     8     9
+    //    │          ┌─────x────┐
+    //    │          │          │
+    //    │          │          │
+    //    │          │          │
+    //    │5         │          │
+    //    ├──────────┘          │
+    //    │           6         │
+    //    │                     │
+    //    │                     │
+    //    │                     │10
+    //    └────x─────x─────x────┴─────►
+    //    4     3      2      1           x
```

Comment:
Unit test case. We expect 1, 2, 3, and 8 to be dropped, resulting in 6 vertices in the end.

Reply:
Before the fix, the floor is checked in the order show above. so it drops 2 and 3, then finds a non convex situation when checking (5, 6, 7) and bails. We end up with 8 vertices.

For the ceiling, the vertices are reversed. So it drops 8 (shown above, 3 in actual order), finds the non convex situation at (7, 6, 5) (shown above, (4, 5, 6 in actual order)) and bails. We end up with 9 vertices.

(I explained the same on #10103 when I reported the defect)

Reply:
Ahhh this makes sense and is a phenomenal demonstration of the problem and test of the routine.  Thanks @jmarrec !

@mjwitte this looks good, but feel free to take a look before this merges.